File: tests/integration/fixtures/go/crypto/certificate/tls/tls.test.go
package main

import (
	"crypto/tls"
)

// TEST-RULE: go.crypto.tls.load-key-pair
// TEST-METADATA: assetType:certificate findingType:certificate_handling operation:other library:crypto/tls api:tls.LoadX509KeyPair materialSource:loaded
func testLoadX509KeyPair() {
	certFile := "server.crt"
	keyFile := "server.key"
	// ruleid: go.crypto.tls.load-key-pair
	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
	if err != nil {
		panic(err)
	}
	_ = cert
}

// TEST-RULE: go.crypto.tls.load-key-pair
// TEST-METADATA: assetType:certificate findingType:certificate_handling operation:other library:crypto/tls api:tls.LoadX509KeyPair materialSource:loaded
func testX509KeyPair() {
	certPEM := []byte("cert")
	keyPEM := []byte("key")
	// ruleid: go.crypto.tls.load-key-pair
	cert, err := tls.X509KeyPair(certPEM, keyPEM)
	if err != nil {
		panic(err)
	}
	_ = cert
}

File: tests/integration/fixtures/rust/native-tls/certificate/tls/tls.test.rs
// TEST-RULE: rust.native-tls.tls.add-root-certificate
// TEST-METADATA: assetType=certificate, findingType=certificate_handling, operation=other, library=native-tls, materialSource=loaded
fn test_add_root_certificate() {
    use native_tls::{Certificate, TlsConnector};
    let builder = TlsConnector::builder();
    let cert_data = b"-----BEGIN CERTIFICATE-----\n...";
    builder.add_root_certificate(Certificate::from_pem(cert_data).unwrap());
}

// TEST-RULE: rust.native-tls.tls.add-root-certificate
// TEST-METADATA: assetType=certificate, findingType=certificate_handling, operation=other, library=native-tls, materialSource=loaded
fn test_add_root_certificate_der() {
    use native_tls::{Certificate, TlsConnector};
    let builder = TlsConnector::builder();
    let cert_data = vec![0x30, 0x82, 0x03, 0x00];
    builder.add_root_certificate(Certificate::from_der(&cert_data).unwrap());
}

// TEST-RULE: rust.native-tls.tls.identity
// TEST-METADATA: assetType=certificate, findingType=certificate_handling, operation=other, library=native-tls, materialSource=loaded
fn test_identity() {
    use native_tls::{Identity, TlsConnector};
    let builder = TlsConnector::builder();
    let pkcs12_data = vec![0x30, 0x82, 0x03, 0x00];
    let password = "password";
    builder.identity(Identity::from_pkcs12(&pkcs12_data, password).unwrap());
}
